### DISCUSSION TITLE
most cost effective sensor coverage

### DISCUSSION THREAD

**Discussion by dansrogers** [mKeRix/room-assistant#881] [Q&A]
Hey there,

What is the most cost effective way to sense a room in a way that doesn't require people to carry a known device to be picked up by room-assistant?

Obviously bluetooth is super cheap and works great as long as folks always have their phone with them.  But ... I like to disconnect or sometimes go to a different room and leave my phone behind for a few minutes.  Or I might have guests and they don't have a device the automation recognizes.

Thermopile sensors as a solution here are *really* expensive. The D6T 8x8 sensor has 5 ft range and 45 degree FOV  This translates to a lot of sensors per room.

**Comment by mKeRix** [comment]
I think the most cost effective setup is to mostly rely on motion sensors with whatever thermopile you can get for the cheapest price pointed at known spots where people may be still for longer than you want your motion timeout to be. Motion sensors can be gotten quite cheap from many different manufacturers, I'd recommend to get some Zigbee or Z-Wave ones. You can also use them from room-assistant, but I'd only do that if you want to use other integrations or software on that device as well, otherwise it would be quite overkill. You can then build your automations around the assumption that a room presence change will only happen after motion was sensed.

Alternatively, as you already mentioned in another thread, occupancy counting can be somewhat cheap as well depending on how many entryways you have. It's not possible in RA yet however. On the room presence Discord I saw someone else building ESP code to do this kind of occupancy counting, but they struggled with inaccuracy and the software going out of sync. They worked with a different kind of a sensor though, not a thermopile.
